blackjax/adaptation/step_size.py

```diff
@@ -257,3 +260,279 @@ def update(rss_state: ReasonableStepSizeState) -> ReasonableStepSizeState:
     rss_state = jax.lax.while_loop(do_continue, update, rss_state)
 
     return rss_state.step_size
+
+
+class MCLMCAdaptationState(NamedTuple):
+    """Tunable parameters for MCLMC"""
+
+    L: float
+    step_size: float
+
+
+
+def ess_corr(x):
```

Comment:
Instead of copying large proportion of code, you should refactor the origin function.

Reply:
Yep, will definitely do that.